scripts/run_backend_tests_test.py

```diff
@@ -1003,3 +1004,37 @@ def mock_run_shell_cmd(*_: str, **__: str) -> str:
 
         self.assertIn('Task result', results[0].messages)
         self.assertEqual(len(results[0].messages), 1)
+
+    def test_skip_install_skips_third_party_installation(self) -> None:
+        from scripts import run_backend_tests
+        mock_third_party_install = unittest.mock.Mock()
+        swap_install = self.swap(
+            install_third_party_libs, 'main', mock_third_party_install)
```

Comment:
Should we use this instead to stay consistent with our other tests?

https://github.com/oppia/oppia/blob/803d2a22ed069342bfda4e2f3571bda80ae491d0/core/tests/test_utils.py#L1606

Side note: Python has a well-developed mock and testing system, but we use our own home-brewed version at Oppia. Do you know why? Is there a reason we can't use the native system, or is this just a historical thing?

Reply:
FYI (re side note only): I think this is tracked at https://github.com/oppia/oppia/issues/18781